src/components/tasks/card/TaskStatusEditMode.tsx

```diff
@@ -1,42 +1,84 @@
+import { useUpdateTaskMutation } from '@/app/services/tasksApi';
+import { useRouter } from 'next/router';
 
 type Props = {
     task: task;
-    updateTask: (changeId: string, changeObject: { status: string }) => void;
+    setEditedTaskDetails: React.Dispatch<React.SetStateAction<CardTaskDetails>>;
 };
 
 // TODO: remove this after fixing the card beautify status
 const beautifyStatus = (status: string) => status.split('_').join(' ');
 const taskStatus = Object.entries(BACKEND_TASK_STATUS);
 
-const TaskStatusEditMode = ({ task, updateTask }: Props) => {
+const TaskStatusEditMode = ({ task, setEditedTaskDetails }: Props) => {
+    const router = useRouter();
+    const { dev } = router.query;
+
+    const [saveStatus, setSaveStatus] = useState('');
+    const [updateTask] = useUpdateTaskMutation();
+
     const onChangeUpdateTaskStatus = ({
         target: { value },
     }: React.ChangeEvent<HTMLSelectElement>) => {
-        updateTask(task.id, {
-            status: value,
+        setSaveStatus(PENDING);
+        setEditedTaskDetails((prev: any) => ({ ...prev, status: value }));
+        const response = updateTask({
+            id: task.id,
+            task: {
+                status: value,
+            },
         });
+
+        response
+            .unwrap()
+            .then((result: any) => {
+                setSaveStatus(SAVED);
+            })
+            .catch((err: { data: { message: string } }) => {
+                setSaveStatus(ERROR_STATUS);
+            })
+            .finally(() => {
+                setTimeout(() => {
+                    setSaveStatus('');
+                }, 3000);
+            });
     };
 
     // TODO: remove this after fixing the card beautify status
     const defaultStatus = task.status.toUpperCase().split(' ').join('_');
 
     return (
-        <label>
-            Status:
-            <select
-                name="status"
-                onChange={onChangeUpdateTaskStatus}
-                defaultValue={defaultStatus}
-            >
-                {taskStatus.map(([name, status]) => (
-                    <option key={status} value={status}>
-                        {beautifyStatus(name)}
-                    </option>
-                ))}
-            </select>
-        </label>
+        <div className={classNames.taskSection}>
+            <label>
+                Status:
+                <select
+                    name="status"
+                    onChange={onChangeUpdateTaskStatus}
+                    defaultValue={defaultStatus}
+                >
+                    {taskStatus.map(([name, status]) => (
+                        <option key={status} value={status}>
+                            {beautifyStatus(name)}
+                        </option>
+                    ))}
+                </select>
+            </label>
+            {dev === 'true' && (
+                <>
+                    {saveStatus === PENDING && <SmallSpinner />}
```

Comment:
maybe take this out in a if else statement and make it a component ?

Reply:
Great suggestion, I have made the separate component for it now.